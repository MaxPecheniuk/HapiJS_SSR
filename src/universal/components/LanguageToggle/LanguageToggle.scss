@import "../../styles/variables";
.header__dropdown{
  position: relative;
  display: inline-block;
  &__btn{
    text-transform: uppercase;
    background-color: $yellowLight;
    color: white;
    padding: 8.5px;
    font-size: 1rem;
    font-weight: bold;
    border: none;
  }
  &__content{
    display: none;
    position: absolute;
    min-width: 70px;
    @extend %cardBackground;
    z-index: 1;
    span{
      cursor: pointer;
      color: black;
      padding: 12px 5px;
      text-decoration: none;
      display: block;
      &:hover{
        background-color: $backgroundLight;
      }
    }
  }
}


.header__dropdown:hover .header__dropdown__content{
  display: block;
}

.header__dropdown:hover .header__dropdown__btn{
  background-color: $yellowDark ;
  border-radius: 3px;
}