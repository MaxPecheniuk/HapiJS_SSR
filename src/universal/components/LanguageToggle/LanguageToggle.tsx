import * as React from 'react';
import * as queryString from 'querystring';
import { match, withRouter } from 'react-router';
// import { History, Location } from 'history';
//
import './LanguageToggle.scss';
import { connect } from 'react-redux';
import { changeLanguageActionCreators } from './actionCreators/LanguageToggle.actionCreators';

interface ILanguageToggleProps {
  match: match;
  location: Location;
  history: History;
  changeLang: (lang: any) => void;
}

const LanguageToggle: React.FunctionComponent<ILanguageToggleProps> = (props: ILanguageToggleProps) => {
  let parsed = queryString.parse(props.location.search.replace('?', ''));
  const changeLang = (language) => () => {
    props.changeLang({language: language, parsed: parsed});
  };
  return (
    <div className="header__dropdown">
      <button className="header__dropdown__btn">{parsed.lang}</button>
      <div className="header__dropdown__content">
        <span
          className="header__dropdown__content_en"
          onClick={changeLang('en')}
        >
          ENG
        </span>
        <span className="header__dropdown__content_ru" onClick={changeLang('ru')}>RUS</span>
      </div>

    </div>
  );
};

const mapStateToProps = (state) => {
  return {language: state.languageReducer.language};
};
const mapDispatchToProps = dispatch => {
  return {
    changeLang: lang => dispatch(changeLanguageActionCreators(lang))
  };
};

export default withRouter(connect<any, any, any>(mapStateToProps, mapDispatchToProps)(LanguageToggle));