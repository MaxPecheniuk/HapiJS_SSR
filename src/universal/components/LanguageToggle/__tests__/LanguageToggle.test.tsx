import * as React from 'react';
import * as enzyme from 'enzyme';
import LanguageToggle from '../LanguageToggle';

describe('LanguageToggle component', () => {
  let component;
  beforeEach(() => {
    component = enzyme.shallow(<LanguageToggle/>);
  })

  it('should render correctly with no props', () => {
    expect(component).toMatchSnapshot();
  });

  it('shallow wrapper instance should be null', () => {
    const instance = component.instance();
    expect(instance).toEqual(null);
  });
});