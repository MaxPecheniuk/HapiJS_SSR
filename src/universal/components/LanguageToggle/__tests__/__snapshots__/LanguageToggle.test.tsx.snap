// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly with no props 1`] = `ShallowWrapper {}`;

exports[`LanguageToggle component should render correctly with no props 1`] = `ShallowWrapper {}`;
